Linux Kernel Survey 20230117

MingcongBai opened the pull request:
Topic Description
-----------------

This topic surveys for kernel (mainline and LTS) stack updates.

Package(s) Affected
-------------------

- [x] `dracut` v059
- [x] `firmware-free` v20230117+debian20221214
- [x] `firmware-nonfree` v20230117+debian20221214
- [x] `kernel-tools` v6.0.19
- [x] `linux+kernel+lemote` v5.4.229
- [x] `linux+kernel+lts` v5.15.88
- [x] `linux+kernel` v6.0.19
- [x] `linux-kernel-lemote` v5.4.229
- [x] `linux-kernel-lts` v5.15.88
- [x] `linux-kernel` v6.0.19
- [x] `pahole` v1.24-1

Security Update?
----------------

Yes?

Build Order
-----------

For most architectures:

```
pahole dracut linux-kernel linux+kernel linux-kernel-lts linux+kernel+lts linux-firmware kernel-tools
```

For `loongson3`:

```
pahole dracut linux-kernel-lemote linux+kernel+lemote linux-firmware kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

FlyGoat committed:
linux-kernel-lemote: update to 5.4.228

Bump version and rebase patches.
Add a patch for NUMA.
Disable CPU AUTOPLUG which is not helpful at all.

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>

MingcongBai committed:
linux-kernel-lts: update to 5.15.88

- Enable Intel SLS (Straight Line Speculation) mitigations.
- Apply upstream defaults for new options.

MingcongBai committed:
linux-kernel{,-lts}: (loongson3) drop configs

Use linux-kernel-lemote instead.

FlyGoat committed:
linux-kernel-lemote: update to 5.4.229

Rebase patches.

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>

FlyGoat committed:
linux-kernel-lemote: Add a patch to fix rx2x00 driver build

Exclude SoC code path on Loongson64.

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>

MingcongBai committed:
linux-kernel-lemote: apply updated upstream defaults

MingcongBai committed:
linux+kernel+lemote: bump to 5.4.229

MingcongBai committed:
dracut: update to 059

- Drop upstreamed patch: 0001-fix-multipath-install-multipathd.socket.patch.
- Include a default dracut.conf (from "$SRCDIR"/dracut.conf), specify multi-
  threaded xz as default compression method.

Fearyncess committed:
linux-kernel-lts: (amd64) raise CONFIG_NR_CPUS value to 8192

Signed-off-by: Lain Yang <fsf@live.com>

MingcongBai committed:
dracut: try using zstd instead

Parallelised xz compression produces unusable initrd.

cthbleachbit committed:
dracut: make triggers noawait

cthbleachbit committed:
plymouth: trigger initrd update, subscribe to theme update

LaszloGombos commented:
I am not very familiar with AOSC, but work on dracut and wanted to drop a quick hint. I am glad to see the update to 059.

Wanted to point out that dracut packaging requires adding the `dracut-version.sh` file explicitly. I believe this is also an issue with the current v57 dracut version (so not specific to the upgrade, but perhaps can be resolved as part of the upgrade).

See https://github.com/dracutdevs/dracut/issues/2169

MingcongBai commented:
@LaszloGombos Thanks for the tip! I'll take care of this soon.

MingcongBai committed:
linux-kernel: (amd64) raise CONFIG_NR_CPUS value to 8192

MingcongBai committed:
linux-kernel: (dev-sort-config.sh) add in-place mode

If file names are passed as arguments, the script will edit the config
in-place. Otherwise, retain the old behavior

MingcongBai committed:
linux-kernel: move dev-sort-config.sh to autobuild/

MingcongBai committed:
linux-kernel: update to 6.2.1

- (all) Enable AppArmor by default.
- (all) Complete CONFIG_LSM by upstream template.
- (all) Run dev-sort-config.sh.
- (all) Enable CONFIG_LEGACY_TIOCSTI to fix issue with McFly. [1]
- Bump zfs PKGBREAK to <= 2.2.0.

FIXME: CONFIG_LEGACY_TIOCSTI has a known security flaw and upstream recommends
disabling by default. [2] But this breaks userspace, which should enrage Linus
Torvalds, and possibly others. Make most of us happy.

Ref [1]: https://github.com/cantino/mcfly/issues/333
Ref [2]: https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=tiocsti

MingcongBai committed:
linux-kernel-lts: update to 6.1.14

- (all) Sync configs from mainline (linux-kernel).
- (all) Enable AppArmor by default.
- (all) Complete CONFIG_LSM by upstream template.
- (all) Run dev-sort-config.sh.

MingcongBai committed:
zfs: update to 2.1.9

MingcongBai committed:
linux-firmware: update to 20230210+debian20210818+1~bpo11+1

Reflect Debian extra firmware version in PKGVER.

MingcongBai committed:
linux-kernel-lemote: update config

- Enable AppArmor by default.
- Complete CONFIG_LSM by upstream template.
- Run dev-sort-config.sh.

MingcongBai committed:
dracut: switch to Git source

This fixes an issue where Dracut does not report the correct version.

Ref: https://github.com/dracutdevs/dracut/issues/2169

MingcongBai committed:
pahole: (Debian backported patches) fix compatibility with Binutils 2.40

Ref: https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1883004.html

MingcongBai committed:
linux+kernel: fix PKGVER

MingcongBai committed:
linux+kernel+lts: fix PKGVER

cthbleachbit committed:
linux-kernel: check whether config was updated by make oldconfig
